testing/tests/DevExpress.ui.widgets.editors/rangeSlider.tests.js

```diff
@@ -1343,3 +1343,81 @@ QUnit.module('valueChanged handler should receive correct event parameter', {
+        this.$element = $('#slider').dxRangeSlider({
+            max: 100,
+            min: 0,
+            start: 20,
+            end: 60,
+            valueChangeMode: 'eventual',
+            onValueChanged: this.valueChangedHandler,
+            tooltip: {
+                enabled: true,
+                showMode: 'always'
+            }
+        });
+        this.instance = this.$element.dxRangeSlider('instance');
+        this.$handle = this.$element.find(`.${SLIDER_HANDLE_CLASS}`);
+        this.leftHandle = this.$handle.eq(0);
+        this.rightHandle = this.$handle.eq(1);
+        this.leftHandleTooltip = this.leftHandle.find(`.${TOOLTIP_CLASS}`);
+        this.rightHandleTooltip = this.rightHandle.find(`.${TOOLTIP_CLASS}`);
+
+        this.range = this.$element.find('.' + SLIDER_RANGE_CLASS);
+
+        this.getLeftTooltipText = () => $.trim(this.leftHandleTooltip.text());
+        this.getRightTooltipText = () => $.trim(this.rightHandleTooltip.text());
+    }
+}, () => {
+    QUnit.test('left slider handle value should not change on swipe with "eventual" valueChangeMode', function(assert) {
+
+        const pointer = pointerMock(this.leftHandle);
+        pointer.start().move(this.range.offset().left).down().move(200);
+        assert.notOk(this.valueChangedHandler.called, 'the onValueChanged is not called');
+
+        pointer.up();
+
+        assert.equal(this.instance.option('start'), 40);
+        assert.ok(this.valueChangedHandler.called, 'the onValueChanged is called');
+    });
+
+    QUnit.test('right slider handle value should not change on swipe with "eventual" valueChangeMode', function(assert) {
+
+        const pointer = pointerMock(this.rightHandle);
+        pointer.start().down(this.rightHandle.offset().left + CONTAINER_MARGIN).move(200);
+        assert.notOk(this.valueChangedHandler.called, 'the onValueChanged is not called');
+
+        pointer.up();
+
+        assert.equal(this.instance.option('end'), 80);
+        assert.ok(this.valueChangedHandler.called, 'the onValueChanged is called');
+    });
+
+    QUnit.test('left slider handle tooltip value should change with "eventual" valueChangeMode', function(assert) {
+        const pointer = pointerMock(this.leftHandle);
+        pointer.start()
+            .down(this.range.offset().left)
+            .move(200);
+
+        assert.strictEqual(this.getLeftTooltipText(), '40');
+
+        pointer.move(400).up();
+
+        assert.strictEqual(this.getLeftTooltipText(), '60');
+    });
+
+    QUnit.test('right slider handle tooltip value should change with "eventual" valueChangeMode', function(assert) {
+        const pointer = pointerMock(this.rightHandle);
+        pointer.start().down(this.rightHandle.offset().left + CONTAINER_MARGIN).move(200);
+
+        assert.strictEqual(this.getRightTooltipText(), '80');
+
+        pointer.move(100).up();
+
+        assert.strictEqual(this.getRightTooltipText(), '90');
+    });
+});
```

Comment:
we should add test on scenario that was broken: when we swipe right handle through left handle

Reply:
Do you think it is also relevant to write a test when is clicked (instead of swipe) for this scenario?

Reply:
Looks like it's already done in this test https://github.com/DevExpress/DevExtreme/pull/22284/files#diff-4494ecb2395df1b3d5abda0b7e429e04da5c693a3480bb59f92cb02077b82307L309